CORE/world_gadgets/router/router_store.rb

```diff
@@ -0,0 +1,69 @@
+
+class RouterStore
+
+  @@stored_specs = {}
+  @@stored_routes = []
+  @@stored_elements = {}
+
+  def self.store_page_spec(pagename, parents)
+    @@stored_specs[pagename] = PageSpec.new(pagename, parents)
+  end
+
+  def self.store_route(source_page, target_page, action, prerequisites)
+    @@stored_routes.push({source_page: source_page,
+                          target_page: target_page,
+                          action: action,
+                          prerequisites: prerequisites})
+  end
+
+  def self.store_element(page, options_hash, action, type)
+    @@stored_elements[page] ||= []
+    @@stored_elements[page].push({options_hash: options_hash, action: action, type: type})
+  end
+
+  def self.store_id_element(page, element_type, id_hash, action)
+    options_hash = { element_type: element_type }.merge(id_hash)
+    self.store_element(page, options_hash, action, :id)
+  end
+
+  def self.store_wait_element(page, element_type, id_hash, action)
+    options_hash = { element_type: element_type }.merge(id_hash)
+    self.store_element(page, options_hash, action, :wait)
+  end
+
+  def self.generate_graph(world)
```

Comment:
It can remain a local variable. I was just thinking of some really simple functions that just return information that this method needs. We can look at this together at some point if you want.